cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -1160,31 +1160,29 @@ SpecificAndLatestVersionKeys LocalVersionedEngine::get_stream_index_map(
     return std::make_pair(specific_versions, latest_versions);
 }
 
+folly::Future<std::pair<std::optional<VariantKey>, std::optional<google::protobuf::Any>>> LocalVersionedEngine::get_metadata_future_impl(std::optional<AtomKey> key){
+    if (key.has_value()){
+        return store()->read_metadata(key.value());
+    }else{
+        return folly::makeFuture(std::make_pair(std::nullopt, std::nullopt));
+    }
+}
+
+folly::Future<std::pair<std::optional<VariantKey>, std::optional<google::protobuf::Any>>> LocalVersionedEngine::get_metadata_future(folly::Future<std::optional<AtomKey>>&& version_fut){
+    return  std::move(version_fut)
+    .thenValue([that = this](std::optional<AtomKey> key){
+        return that->get_metadata_future_impl(key);
```

Comment:
that = this is a pattern that is no longer needed, we can just capture this
```suggestion
    .thenValue([this](std::optional<AtomKey> key){
        return get_metadata_future_impl(key);
```